src/server/tiered_storage.cc

```diff
@@ -139,40 +140,44 @@ TieredStorageV2::~TieredStorageV2() {
 }
 
 error_code TieredStorageV2::Open(string_view path) {
-  return op_manager_->Open(path);
+  return op_manager_->Open(absl::StrCat(path, ProactorBase::me()->GetPoolIndex()));
 }
 
 void TieredStorageV2::Close() {
   op_manager_->Close();
 }
 
-util::fb2::Future<string> TieredStorageV2::Read(string_view key, const PrimeValue& value) {
+util::fb2::Future<string> TieredStorageV2::Read(DbIndex dbid, string_view key,
+                                                const PrimeValue& value) {
   DCHECK(value.IsExternal());
   util::fb2::Future<string> future;
-  op_manager_->Enqueue(key, value.GetExternalSlice(), [future](string* value) mutable {
+  auto cb = [future](string* value) mutable {
     future.Resolve(*value);
     return false;
-  });
+  };
+  op_manager_->Enqueue(std::make_pair(dbid, key), value.GetExternalSlice(), std::move(cb));
```

Comment:
nit: I find it more robust to use `KeyRef (...)` instead of `make_pair` because if they type representation changes you do not need to replace everything.